### ISSUE TITLE
FSA option is in a poor location.

### ISSUE THREAD

**Issue by CyrusNajmabadi** [dotnet/roslyn#14994]
The option for FSA is both hard to find, and hard to understand.  Right now it appears here:

![image](https://cloud.githubusercontent.com/assets/4564579/20021124/b4f711ee-a291-11e6-9bb2-804a2560359d.png)

As part of this entire set of options:

![image](https://cloud.githubusercontent.com/assets/4564579/20021138/bfeb32ec-a291-11e6-89c3-08435646271c.png)

This is too important an option to bury in that section of the options list.  IMO, this should be a top level item, and it should have text explaining what it does.  Perhaps something more user friendly like "Show suggestions, warnings and errors for files that are not open for editing."

**Comment by CyrusNajmabadi** [comment]
"Full solution analysis" doesn't mean anything to me as a user.  What does 'analysis' mean?  What does it mean if i turn this off/on?  Feature options should explain what the end result is for the user, and what the consequences are (if there are any).

**Comment by CyrusNajmabadi** [comment]
Tagging @srivatsn

**Comment by srivatsn** [comment]
Tagging @morrisjoe @kuhlenh

**Comment by miloush** [comment]
"Full solution analysis" means "slow" to me, regardless of what "analysis" mean.

If it is for open files only (as opposed to other projects), that should definitely be mentioned.

**Comment by CyrusNajmabadi** [comment]
@srivatsn @kuhlenh @morrisjoe Are we doing anything about this?  Right now the experience is quite poor.  This feature ends up disabling IDE features with nothing messages to he user to make that clear.  According to Heejae, this is ok since people will go turn the option on if they want teh features.  However, i don't believe that argument holds because:

1. Users aren't even aware the features aren't there.  Unless someone tells them, there's no way for the user to know the feature even exists.  This is because, with FSA off, the feature never lights up for the user *Ever*.  So there is no way for them to know the feature is gone.
2.  The option is specifically *underdocumented*.  i.e. we do not explain to the user what 'full solution analysis' means.  And, even if we tried to, how would we explain this issue.  "When full solution analysis is off, CompilationEndActions don't run.  So analyzer that depend on them will simply be broken and will never report results".   

There is a way to solve this (which we have discussed at length in the past, but which i've seen *no* forward progress on).  We change teh option from "full solution analysis" to "Produce diagnostics for open files only".  That becomes much clearer to the user what the option means, and it directs how we implement the actual feature".  Even if this option was enabled, we'd still need to run COmpilationEnd actions for projects that had open files.  Then we'd actually produce the diagnostics for the files the user has open and they would be able to benefit from these features.

--

As mentioned before we have little runway for this.  So i'm highly concerned that this work won't be done, even though we've been talking about this for i believe 9+ months at this point.
